apps/image-editor/src/js/action.js

```diff
@@ -405,6 +406,109 @@ export default {
     );
   },
 
+  /**
+   * Resize Action
+   * @returns {Object} actions for ui resize
+   * @private
+   */
+  _resizeAction() {
+    return extend(
+      {
+        getCurrentDimensions: () => this._graphics.getCurrentDimensions(),
+        preview: (actor, value, lockState) => {
+          const currentDimensions = this._graphics.getCurrentDimensions();
+          const calcAspectRatio = () => currentDimensions.width / currentDimensions.height;
+
+          let dimensions = {};
+          switch(actor) {
+            case 'width':
+              dimensions.width = value;
+              if (lockState) {
+                dimensions.height = value / calcAspectRatio();
+              } else {
+                dimensions.height = currentDimensions.height;
+              }
+              break;
+            case 'height':
+              dimensions.height = value;
+              if (lockState) {
+                dimensions.width = value * calcAspectRatio();
+              } else {
+                dimensions.width = currentDimensions.width;
+              }
+              break;
+            default:
+              dimensions = currentDimensions;
+          }
+
+          this._graphics.resize(dimensions).then(() => {
+            this.ui.resizeEditor();
+          });
+
+          if (lockState) {
+            this.ui.resize.setWidthValue(dimensions.width);
+            this.ui.resize.setHeightValue(dimensions.height);
+          }
+        },
+        lockAspectRatio: (lockState, min, max) => {
+          const { width, height } = this._graphics.getCurrentDimensions();
+          const aspectRatio = width / height;
+          if (lockState) {
+            if (width > height) {
+              const pMax = max / aspectRatio;
+              const pMin = min * aspectRatio;
+              this.ui.resize.setMaxHeightValue(pMax < max ? pMax : max);
+              this.ui.resize.setMinWidthValue(pMin > min ? pMin : min);
+              this.ui.resize.setMaxWidthValue(max);
+              this.ui.resize.setMinHeightValue(min);
+            } else {
+              const pMax = max * aspectRatio;
+              const pMin = min / aspectRatio;
+              this.ui.resize.setMaxWidthValue(pMax < max ? pMax : max);
+              this.ui.resize.setMinHeightValue(pMin > min ? pMin : min);
+              this.ui.resize.setMaxHeightValue(max);
+              this.ui.resize.setMinWidthValue(min);
+            }
+          } else {
+            this.ui.resize.setMaxHeightValue(max);
+            this.ui.resize.setMaxWidthValue(max);
+            this.ui.resize.setMinWidthValue(min);
+            this.ui.resize.setMinHeightValue(min);
```

Comment:
How about combining 4 functions that set `min` and `max` values? If you don't have to be called alone, you'd better put it together.
```js
// example
this.ui.resize.setLimit({
  minWidth: min,
  minHeight: min,
  maxWidth: max,
  maxHeight: max,
});
```
If you have a chance to be called alone, you can use it as it is. :)